bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/handler/SoulissGatewayHandler.java

```diff
@@ -0,0 +1,331 @@
+
+import java.math.BigDecimal;
+import java.net.DatagramSocket;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.souliss.SoulissBindingConstants;
+import org.openhab.binding.souliss.SoulissBindingUDPConstants;
+import org.openhab.binding.souliss.internal.SoulissDatagramSocketFactory;
+import org.openhab.binding.souliss.internal.protocol.SoulissBindingNetworkParameters;
+import org.openhab.binding.souliss.internal.protocol.SoulissBindingSendDispatcherJob;
+import org.openhab.binding.souliss.internal.protocol.SoulissBindingUDPServerJob;
+import org.openhab.binding.souliss.internal.protocol.SoulissCommonCommands;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SoulissGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Tonino Fazio - Initial contribution
+ * @author Luca Calcaterra - Refactor for OH3
+ */
+@NonNullByDefault
+public class SoulissGatewayHandler extends BaseBridgeHandler {
+
+    private Logger logger = LoggerFactory.getLogger(SoulissGatewayHandler.class);
+    public @Nullable DatagramSocket datagramSocketDefaultPort;
+    private @Nullable SoulissBindingUDPServerJob udpServerDefaultPortRunnableClass;
+    private SoulissCommonCommands soulissCommands = new SoulissCommonCommands();
+
+    boolean bGatewayDetected = false;
+
+    private @NonNullByDefault({}) Configuration gwConfigurationMap;
+
+    public int pingRefreshInterval;
+    public int subscriptionRefreshInterval;
+    public boolean thereIsAThingDetection = true;
+    public int healthRefreshInterval;
+    public int sendRefreshInterval;
+    public int sendTimeoutToRequeue;
+    public int sendTimeoutToRemovePacket;
+    private Bridge bridge;
+    public int preferredLocalPort;
+    public int soulissGatewayPort;
+    public byte userIndex;
+    public byte nodeIndex;
+    public String ipAddressOnLAN = "";
+    private int nodes;
+    private int maxTypicalXnode;
+    private int countPingKo = 0;
+    // private int maxnodes = 0;
+    // private int maxrequests = 0;
+
+    public SoulissGatewayHandler(Bridge _bridge) {
+        super(_bridge);
+        bridge = _bridge;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initializing server handler for thingGeneric {}", getThing());
+
+        gwConfigurationMap = bridge.getConfiguration();
```

Comment:
If you create a SoulissBridgeConfig class for holding the config you can do this. The advantage is the config is then cached so you don't need to reparse it all the time and no need to create local variables as they are contained in the SoulissBridgeConfig class.

SoulissBridgeConfig soulissBridgeConfig = getConfig().as(SoulissBridgeConfig.class);

if(soulissBridgeConfig.preferredLocalPort>0){
//code here
}

Most bindings do this so check them out for examples.

Reply:
Understand you... anyway i've no time to refactor in this moment . Changes reflects on other classes..

Reply:
done